### ISSUE TITLE
Inkstitch freezing when 'Params' settings applied / No image on "Print" Screen

### ISSUE THREAD

**Issue by ALPHAMOTIF** [inkstitch/inkstitch#474]
Hello! Very new to inkstitch however I am loving the program, thank you for empowering poor artists such as myself to create! Thank you for this gem!  I do have two small issues
1.) On designs over around 11,000 stitches when I adjust more than one "Params" settings and click apply, a double pop up window appears and inkstitch seems to freeze completely (waited it out for 15 minutes with no luck).  One of the pop ups is the normal "params running" message however the other is just a blank box that will not go away.  The program completely locks up when this happens.  I am able to get around this by editing one letter, one param setting at a time but this is rather slow.  This did not seem to happen until the stitch count got very high.
2.) When I click the "Print" extension I just get a pop up white screen with nothing on it, If ive read correctly there is supposed to be a stitch out or something on this page?  Any help would be awesome!
I have installed this program on a Mac operating on Mojave 10.4 using Xquartz to run the program (no idea if this is relevant but I thought it may help)

**Comment by ALPHAMOTIF** [comment]
<img width="1073" alt="Screen Shot 2019-06-12 at 12 54 21 AM" src="https://user-images.githubusercontent.com/51738504/59333190-aad72200-8cac-11e9-932b-c880d8c74eef.png">

Here is an example of issue 1 in action!

**Comment by lexelby** [comment]
Hi @ALPHAMOTIF, thanks for trying out Ink/Stitch, and I'm glad you're enjoying it!

Ink/Stitch can get slower on bigger designs, and that's something I want to improve in the future.  It sounds like you may have found a bug, though.  Any chance you might be able to share your SVG?

**Comment by ALPHAMOTIF** [comment]
Ah that's kind of what I figured at this point!  However I still cannot seem to get the print screen to work, here is the file I made, it sews out beautifully!
[EMBRO PROTO MASTER FINAL copy.svg.zip](https://github.com/inkstitch/inkstitch/files/3309218/EMBRO.PROTO.MASTER.FINAL.copy.svg.zip)

**Comment by ALPHAMOTIF** [comment]
**I should mention the print screen does not work for me during any designs not just this one  (once again, not a big deal for me at all) Also maybe this is just user error but I noticed for some reason if you copy and paste certain elements things get a little funky, I.E. words with multiples of letters like "savage" if i duplicate the first "A" and use that for the second one it seems to trigger some weird behavior

**Comment by lexelby** [comment]
Thanks!  I'll check that out when I'm in front of my computer.  I forgot to ask, what operating system are you using?  And what version of Ink/Stitch?  If you don't remember the version, an idea of around when you downloaded it might also help.

As far as weird behavior with copy-pasted objects, can you share some detail on what happens?  A screenshot or screen recording can also help.

Thanks!

**Comment by ALPHAMOTIF** [comment]
I’m using what I think is the latest version of Ink/Stitch (downloaded within the last month) and operating it on Mojave for Mac, I’ll get some more details shortly!

TAYLOR B.
ALPHAMOTIF
info@alphamotif.net

**Comment by ALPHAMOTIF** [comment]
Heres a screen capture of the recent issue Ive been having, even when doing one section of each letter at a time

[INKSTITCH CRASH PARAMS small.mov.zip](https://github.com/inkstitch/inkstitch/files/3342774/INKSTITCH.CRASH.PARAMS.small.mov.zip)

**Comment by lexelby** [comment]
Ugh, I bet that's annoying!  Thank you so much for taking the time to post that video, it really helped me to understand your issue.

I have a hunch what's going on here.  I think it's a bug in the little simulation view.  That extra window you see is the simulation view trying to open back up.

Try this: before you click "Apply", close the right-side window that has the simulation in it.  Does that prevent the freeze?

As to the print issue, that may be something else entirely.  A video would really help here as well, if you have the time.

**Comment by ALPHAMOTIF** [comment]
Finally had a second to sit down and get more info on this!  So to pick up where we left off, closing the simulation view first doesn't seem to help the crashing of large files, same issue seems to happen - Ive learned to just save early and save often to avoid this being a real problem.  The print preview screen is still a no go for me, Ive attached a video of the issue, maybe this will help!

[print failure.mov.zip](https://github.com/inkstitch/inkstitch/files/3414605/print.failure.mov.zip)

**Comment by ALPHAMOTIF** [comment]
Hello! Just touching base on this issue a bit more!  It seems to be getting a bit worse, if I come back to an old file and try to change the params of more than one column at once the entire program crashes like before, any tips on how to fix this?  Its happening with files that are quite small

**Comment by lexelby** [comment]
Oh sorry, I lost track of this one.  Hmm.  I'll have to dig in further and figure out what's crashing Params.  I'm sorry this is causing so many problems for you!

That movie of the print PDF issue is really helpful.  That tells me that the electron UI _is_ opening a window, which is a great first step.

The next step is to try to figure out why none of the content is loading.  Electron has a "developer tools" option, and I'd like to see if you can open it up.  Try looking in the "inkstitch-gui" menu at the top to see if you can find it there.  If not, try control-shift-I, or apple-shift-I or something like that.  If all that fails, let me know and I'll build a version of Ink/Stitch that opens the Developer Tools automatically.

Once you get the developer tools open, if you look in the Console tab, there might be an error message there that could give us a clue.

**Comment by ALPHAMOTIF** [comment]
I don't think I got the print screen to work! Also I'm still having a bit of trouble figuring out why I cannot make specific color files,  If I make a black and white embroidery file it comes out random colors like yellow and blue on the saved file, just a pain when assigning needle colors on the machine!

**Comment by lexelby** [comment]
You did have a blank window appear in the video, that's what I meant :)

As to colors, I see we're talking at the same time in two different places. ;)  Blue and yellow, that definitely sounds wrong.  I would expect that behavior from a DST, but not a PES.  I'll take a look.

**Comment by ALPHAMOTIF** [comment]
Here’s what I’m seeing for the input and output here, it is chopping it up into two different colors but not the right colors 😅



TAYLOR B.
ALPHAMOTIF
info@alphamotif.net

**Comment by ALPHAMOTIF** [comment]
This time I got gold and green 😂😅

TAYLOR B.
ALPHAMOTIF
info@alphamotif.net

**Comment by lexelby** [comment]
If you attached an image to an email reply, I don't think GitHub got it...

**Comment by ALPHAMOTIF** [comment]
haha sorry about that! It was just screen shots of the Black and white design turning into a green and yellow one on export!  I was up pretty late trying to work on getting the colors to work right XD

**Comment by lexelby** [comment]
It's really weird, I can't think of a reason that would be happening.  This is definitely with PES format?  Does your machine accept a different format?

**Comment by ALPHAMOTIF** [comment]
It accepts DST format however these are usually void of color info.  Also about the GUI print screen issue, I am having trouble figuring out what to do to get the developer tools to open, when I click print then the blank screen pops up the only thing on the GUI menu is "quit"

**Comment by ALPHAMOTIF** [comment]
I just tried clearing my old directory and doing a complete re-install of the software, I can confirm that for some reason PES files will not allow me to choose output colors of the design, this time I exported a black and white logo and got yellow and tan 
<img width="955" alt="Screen Shot 2019-08-20 at 6 17 21 PM" src="https://user-images.githubusercontent.com/51738504/63394975-fece4900-c376-11e9-9de3-1b0711391c84.png">
![Uploading Screen Shot 2019-08-20 at 6.18.47 PM.png…]()

**Comment by ALPHAMOTIF** [comment]
<img width="838" alt="Screen Shot 2019-08-20 at 6 18 47 PM" src="https://user-images.githubusercontent.com/51738504/63395563-f414b380-c378-11e9-8583-094514625f00.png">

**Comment by ALPHAMOTIF** [comment]
I think I found the solution!  I tried just saving one of these designs alone as a PES to see how it imported.  The single box imported with the colors correct so I doubled back and grouped all similar elements and tried that - worked like a charm!  So it seems like as long as I always remember to group objects by color I should be able to control the output just fine!  Now if only I could figure out that print screen XD
